Adding LTM Auth endpoint

wojtek0806 opened the pull request:
Problem:
LTM auth endpoints were missing from the SDK, despite that some of the LTM auth features were discontinued(due to move to APM), this is still used and is a valid rest point.

Analysis:
Added LTM auth endpoint to SDK, some of the features require a retured client auth module addon (still available on some older platforms, i.e 1600, 3600, 3900 etc.) i have tested those against hardware unit, however they are skipped by default, and were added for completeness.

Tests:
Flake8
Functional Tests
Unit Tests

Files Added/Changed:

f5-common-python/f5/bigip/tm/ltm/init.py
f5-common-python/f5/bigip/tm/ltm/auth.py
f5-common-python/f5/bigip/tm/ltm/test/test_auth.py
f5-common-python/test/functional/tm/ltm/test_auth.py

wojtek0806 committed:
Fixes #410

Problem:
LTM auth endpoints were missing from the SDK, despite that some of the LTM auth features were discontinued(due to move to APM), this is still used and is a valid rest point.

Analysis:
Added LTM auth endpoint to SDK, some of the features require a retured client auth module addon (still available on some older platforms, i.e 1600, 3600, 3900 etc.) i have tested those against hardware unit, however they are skipped by default, and were added for completeness.

Tests:
Flake8
Functional Tests
Unit Tests

Files Added/Changed:

f5-common-python/f5/bigip/tm/ltm/init.py
f5-common-python/f5/bigip/tm/ltm/auth.py
f5-common-python/f5/bigip/tm/ltm/test/test_auth.py
f5-common-python/test/functional/tm/ltm/test_auth.py

wojtek0806 committed:
Fixes #410

Problem:
LTM auth endpoints were missing from the SDK, despite that some of the LTM auth features were discontinued(due to move to APM), this is still used and is a valid rest point.

Analysis:
Added LTM auth endpoint to SDK, some of the features require a retured client auth module addon (still available on some older platforms, i.e 1600, 3600, 3900 etc.) i have tested those against hardware unit, however they are skipped by default, and were added for completeness.

Tests:
Flake8
Functional Tests
Unit Tests

Files Added/Changed:

f5-common-python/f5/bigip/tm/ltm/init.py
f5-common-python/f5/bigip/tm/ltm/auth.py
f5-common-python/f5/bigip/tm/ltm/test/test_auth.py
f5-common-python/test/functional/tm/ltm/test_auth.py

caphrim007 commented on test/functional/tm/ltm/test_auth.py:
```diff
@@ -0,0 +1,345 @@
+        authres.__dict__['_meta_data'] = meta_data
+        authres.modify(description='MODIFIED')
+        desc = 'description'
+        for k, v in authres.__dict__.items():
```
Can you change this to iteritems(authres.**dict**) for python3 and add an "from six import iteritems" at the top of the file

caphrim007 commented on test/functional/tm/ltm/test_auth.py:
```diff
@@ -0,0 +1,345 @@
+        authres.__dict__['_meta_data'] = meta_data
+        authres.modify(idleTimeout=100)
+        desc = 'idleTimeout'
+        for k, v in authres.__dict__.items():
```
iteritems(authres.**dict**)

caphrim007 commented on f5/bigip/tm/ltm/test/test_auth.py:
```diff
@@ -0,0 +1,179 @@
+        test_meta = rc._meta_data['attribute_registry']
+        test_meta2 = rc._meta_data['allowed_lazy_attributes']
+        kind = self.authkinds[self.urielementname()]
+        assert kind in test_meta.keys()
```
change this to list(iterkeys(test_meta)) and include "from six import iterkeys" at the top of the file for python 3

caphrim007 commented:
lgtm after the noted changes are addressed

wojtek0806 committed:
python3 corrections

wojtek0806 committed:
python3 corrections

wojtek0806 committed:
Add LTM Traffic Class Endpoint (#685)

* Fixes #409

Problem:
LTM traffic-class endpoint was missing from the SDK

Analysis:
Added LTM traffic-class endpoint to SDK

Tests:
Flake8
Functional Tests
Unit Tests

Files Added/Changed:

f5-common-python/f5/bigip/tm/ltm/init.py
f5-common-python/f5/bigip/tm/ltm/traffic_class.py
f5-common-python/f5/bigip/tm/ltm/test/test_traffic_class.py
f5-common-python/test/functional/tm/ltm/test_traffic_class.py

* python3 corrections

caphrim007 commented:
@wojtek0806 i believe this needs a rebase as I dont see the previously merged traffic_class thing in it

wojtek0806 committed:
Merge remote-tracking branch 'origin/ltm_auth' into ltm_auth
